op-node/rollup/derive/batches_espresso_test.go

```diff
@@ -12,25 +12,20 @@ import (
 	"github.com/ethereum-optimism/optimism/op-service/espresso"
 	"github.com/ethereum-optimism/optimism/op-service/eth"
 	"github.com/ethereum/go-ethereum/common/hexutil"
-	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
 )
 
-type EspressoValidBatchTestCase struct {
-	Name       string
-	L1Blocks   []eth.L1BlockRef
-	L2SafeHead eth.L2BlockRef
-	Batch      BatchWithL1InclusionBlock
-	Expected   BatchValidity
-}
-
 type mockHotShotProvider struct {
 }
 
 func (m *mockHotShotProvider) verifyHeaders(headers []espresso.Header, height uint64) error {
 	return nil
 }
 
+func (m *mockHotShotProvider) getHeadersFromHeight(firstBlockHeight uint64, numHeaders uint64) ([]espresso.Header, error) {
+	return []espresso.Header{}, nil
```

Comment:
Why not give `mockHotShotProvider` a `[]Header` field? We can configure this in each test based on the HotShot state we want to simulate, and then we can actually implement this function and `verifyHeaders`. In particular I think it would be good for `verifyHeaders` to have a real implementation so we can test the case where header validation fails.

Reply:
Done 👍 , added todos to handle invalid headers when I implement #50